azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -518,6 +520,51 @@ private void handleDownloadProject(HttpServletRequest req,
 
   }
 
+    /**
+     * validate readiness of a project and user permission and use
+     * projectManager to purge the project if things looks good
+     **/
+    private void handlePurgeProject(HttpServletRequest req,
+        HttpServletResponse resp, Session session) throws ServletException,
+        IOException {
+        User user = session.getUser();
+        String projectName = getParam(req, "project");
+        HashMap<String, Object> ret = new HashMap<String, Object>();
+        boolean valid = true;
+
+        if (projectManager.isActiveProject(projectName)) {
+            ret.put("error", "Project " + projectName
+                + " should be deleted before purging");
+            valid = false;
+        }
+
+        try {
+            Project project = null;
+
+            // project is already deleted
+            if (valid) {
+                project = projectManager.getProject(projectName);
+                // only eligible users can purge a project
+                if (project == null
+                    || !hasPermission(project, user, Type.ADMIN)) {
```

Comment:
Looks like this if statement contains 2 possible scenarios.  One is project already deleted.  The other is user does not have admin permission.  Should they be sending out different error messages?